launch/launch/utilities/type_utils.py

```diff
@@ -314,7 +314,7 @@ def coerce_list(
     ensure_argument_type(value, list, 'value', 'coerce_list')
     output = [coerce_to_type(i, data_type, can_be_str=can_be_str) for i in value]
     if not is_instance_of_valid_type(output, can_be_str=can_be_str):
-        raise ValueError(f'cannot convert value to {data_type}. Got value=`{value}`')
+        raise ValueError(f'cannot convert value to {data_type.__name__}. Got value=`{value}`')
```

Comment:
Looks like `data_type` can be `None` here and below, see the test failures: https://build.ros2.org/job/Rpr__launch__ubuntu_noble_amd64/283/consoleFull#console-section-59